parmed/amber/parameters.py

```diff
@@ -94,7 +94,13 @@ def id_format(filename):
         is_fmt : bool
             True if it is an Amber-style parameter file. False otherwise.
         """
-        with closing(genopen(filename, 'r')) as f:
+        if isinstance(filename, string_types):
```

Comment:
Also, this hard-codes an assumption that `filename.tell()` returns 0 in the beginning.